src/Nethermind/Nethermind.Synchronization/ParallelSync/SyncDispatcher.cs

```diff
@@ -272,5 +289,22 @@ private void UpdateState(SyncFeedState state)
                 }
             }
         }
+
+        public ValueTask DisposeAsync()
+        {
+            if (_disposed)
+            {
+                return ValueTask.CompletedTask;
+            }
+            _disposed = true;
+
+            _activeTasks.Signal();
```

Comment:
Can the event start with count 0 and we can drop this signal?

Reply:
No, if it start with 0 once something add to it, then back to 0, it will throw on new task.